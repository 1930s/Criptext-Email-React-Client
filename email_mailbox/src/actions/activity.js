import { Activity } from './types';

export const startLoadSync = () => {
  return {
    type: Activity.START_LOAD_SYNC
  };
};

export const stopLoadSync = () => {
  return {
    type: Activity.STOP_LOAD_SYNC
  };
};

export const startLoadThread = () => {
  return {
    type: Activity.START_LOAD_THREAD
  };
};

export const stopLoadThread = () => {
  return {
    type: Activity.STOP_LOAD_THREAD
  };
};
