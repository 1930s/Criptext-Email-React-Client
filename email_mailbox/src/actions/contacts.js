import { Contact } from './types';
import * as db from '../utils/electronInterface';

export const addContacts = contacts => {
  return {
    type: Contact.ADD_BATCH,
    contacts: contacts
  };
};

export const loadContacts = ids => {
  return async dispatch => {
    try {
      const response = await db.getContactByIds(ids);
      const contacts = {};
      response.forEach(element => {
        contacts[element.id] = element;
      });
      dispatch(addContacts(contacts));
    } catch (e) {
      // TO DO
    }
  };
};
