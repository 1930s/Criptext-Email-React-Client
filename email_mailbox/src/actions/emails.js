import { Email } from './types';
import {
  getEmailsByThreadId,
  setMuteEmailById
} from '../utils/electronInterface';

export const addEmails = emails => {
  return {
    type: Email.ADD_BATCH,
    emails: emails
  };
};

export const muteNotifications = emailId => {
  return {
    type: Email.MUTE,
    targetEmail: emailId
  };
};

export const loadEmails = threadId => {
  return async dispatch => {
    try {
      const response = await getEmailsByThreadId(threadId);
      const emails = {};
      response.forEach(element => {
        emails[element.id] = element;
      });
      dispatch(addEmails(emails));
    } catch (e) {
      // TO DO
    }
  };
};

export const muteEmailById = feed => {
  return async dispatch => {
    try {
      const valueToSet = feed.get('isMuted') === 1 ? false : true;
      await setMuteEmailById(feed.get('emailId'), valueToSet);
      dispatch(muteNotifications(feed.get('emailId')));
    } catch (e) {
      // To do
    }
  };
};
