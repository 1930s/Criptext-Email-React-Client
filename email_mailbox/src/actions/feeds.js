import { Feed, Thread } from './types';

export const addFeeds = feeds => {
  return {
    type: Feed.ADD_BATCH,
    feeds: feeds
  };
};

export const muteNotifications = threadId => {
  return {
    type: Thread.MUTE_THREAD,
    targetThread: threadId
  };
};

export const toggleMuteFeed = feedId => {
  return {
    type: Feed.TOGGLE_MUTE,
    targetFeed: feedId
  };
};

export const removeFeed = feedId => {
  return {
    type: Feed.REMOVE,
    targetFeed: feedId
  };
};

export const selectFeed = feedId => {
  return {
    type: Feed.SELECT,
    selectedFeed: feedId
  };
};

export const loadFeeds = () => {
  return async dispatch => {
    try {
      const response = await fetch('/feeds.json');
      const json = await response.json();
      const feeds = json.feeds;
      dispatch(addFeeds(feeds));
    } catch (e) {
      // TO DO
    }
  };
};
