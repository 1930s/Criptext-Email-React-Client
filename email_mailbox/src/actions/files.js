import { File } from './types';
import * as db from '../utils/electronInterface';

export const addFiles = files => {
  return {
    type: File.ADD_BATCH,
    files
  };
};

export const loadFiles = tokens => {
  return async dispatch => {
    try {
      const response = await db.getFilesByTokens(tokens);
      const files = response.reduce(
        (result, file) => ({
          ...result,
          [file.token]: file
        }),
        {}
      );
      dispatch(addFiles(files));
    } catch (e) {
      // TO DO
    }
  };
};
