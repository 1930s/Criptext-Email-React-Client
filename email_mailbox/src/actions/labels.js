import { Label } from './types';
import { getAllLabels } from '../utils/electronInterface';
const inLabel = 10;

export const addLabels = labels => {
  return {
    type: Label.ADD_BATCH,
    labels: labels
  };
};

export const addLabel = label => {
  const labels = {
    inLabel: {
      id: inLabel,
      text: label
    }
  };
  return {
    type: Label.ADD_BATCH,
    labels: labels
  };
};

export const loadLabels = () => {
  return async dispatch => {
    try {
      const response = await getAllLabels();
      const labels = {};
      response.forEach(element => {
        labels[element.id] = {
          id: element.id,
          color: element.color,
          text: element.text
        };
      });
      dispatch(addLabels(labels));
    } catch (e) {
      // TO DO
    }
  };
};
