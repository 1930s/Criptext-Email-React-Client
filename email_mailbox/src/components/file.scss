@import "./../styles/index.scss";

.file-container{
  align-items: center;
  border: 1px solid #e7e5e5;
  border-radius: 6px;
  display: flex;
  height: 38px;
  margin-right: 20px;
  position: relative;
  width: 256px;

  &:hover{
    background-color: #f7f6f6;

    .file-detail-size{
      color: #f7f6f6;
    }
    button{
      display: flex;
    }
  }

  button{
    background: transparent;
    display: none;
    position: absolute;
    right: 20px;
    
    i{
      color: #89909a;
      font-size: 13px;
    }
  }
}

.file-content-icon{
  margin-left: 12px;
  position: relative;

  >div{
    align-items: center;
    border-radius: 4px;
    display: flex;
    height: 22px;
    width: 22px;

    i{
      color: white;
      margin: 0 auto;
    }
  }

  .icon-container-file-default{
    background-color: #4a90e2;
  }
  
}

.file-content-detail{
  display: flex;
  flex-grow: 1;
  margin-left: 10px;
  padding-right: 14px;

  .file-detail-name{
    display: flex;
    flex-grow: 1;
    font-size: 14px;
    font-weight: 700;
  }

  .file-detail-size{
    color: #9b9b9b;
    flex-grow: 0;
    font-size: 13px;
  }
}