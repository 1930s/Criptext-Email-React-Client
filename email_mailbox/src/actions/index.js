import { addUsers, addUser } from './users';
import {
  addThreads,
  addThreadLabel,
  addThreadsLabel,
  filterThreadsByUnread,
  loadThreads,
  markThreadsRead,
  moveThreads,
  multiSelectThread,
  muteNotifications,
  selectThread,
  removeThread,
  removeThreadLabel,
  deselectThreads,
  searchThreads,
  selectThreads,
  removeThreads,
  removeThreadsLabel
} from './threads';
import { addEmails, loadEmails } from './emails';
import { addLabels, addLabel, loadLabels } from './labels';
import {
  addFeeds,
  loadFeeds,
  removeFeed,
  selectFeed,
  toggleMuteFeed
} from './feeds';
import { setThreads, loadThreadsSuggestions } from './threadsSuggestions'

export {
  addEmails,
  addFeeds,
  addLabel,
  addLabels,
  addThreads,
  addThreadLabel,
  addThreadsLabel,
  addUser,
  addUsers,
  filterThreadsByUnread,
  loadEmails,
  loadFeeds,
  loadLabels,
  loadThreads,
  loadThreadsSuggestions,
  markThreadsRead,
  moveThreads,
  multiSelectThread,
  muteNotifications,
  removeThread,
  removeThreadLabel,
  deselectThreads,
  searchThreads,
  selectThreads,
  removeFeed,
  removeThreads,
  removeThreadsLabel,
  selectThread,
  selectFeed,
  setThreads,
  toggleMuteFeed
};
