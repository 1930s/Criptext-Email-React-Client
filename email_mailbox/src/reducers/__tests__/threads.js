/* eslint-env node, jest */

import threadsReducer from './../threads';
import * as actions from './../../actions/index';
import file from './../../../public/threads.json';

jest.mock('./../../utils/electronInterface');
jest.mock('./../../utils/electronEventInterface');

const myThreads = file.threads;

describe('Set thread state by actions', () => {
  const threads = [myThreads[0]];
  const manyThreads = [myThreads[0], myThreads[1]];

  function initState(threads) {
    return threadsReducer(undefined, actions.addThreads(threads));
  }

  it('should add threads to state, action[ADD_BATCH]', () => {
    expect(initState(manyThreads)).toMatchSnapshot();
  });

  it('should set thread param: unread, action[UPDATE_UNREAD_THREAD]', () => {
    const state = initState(threads);
    const params = {
      id: 1,
      unread: false
    };
    const action = actions.updateUnreadThread(params);
    const newState = threadsReducer(state, action);
    const emailUpdated = newState.get('0');
    const unread = emailUpdated.get('unread');
    expect(unread).toBe(false);
  });

  it('should set thread param: status, action[UPDATE_STATUS]', () => {
    const state = initState(threads);
    const threadId = 1;
    const newStatus = 2;
    const action = actions.updateStatusThread(threadId, newStatus);
    const newState = threadsReducer(state, action);
    const emailUpdated = newState.get('0');
    const status = emailUpdated.get('status');
    expect(status).toBe(newStatus);
  });

  it('should not set thread param status because is undefined, action[UPDATE_STATUS]', () => {
    const state = initState(threads);
    const threadId = 1;
    const badStatus = undefined;
    const action = actions.updateStatusThread(threadId, badStatus);
    const newState = threadsReducer(state, action);
    const emailUpdated = newState.get('0');
    const status = emailUpdated.get('status');
    expect(status).not.toBe(badStatus);
  });
});
