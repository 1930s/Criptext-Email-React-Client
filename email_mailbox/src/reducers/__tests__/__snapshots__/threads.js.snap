// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`thread actions should add label 6 to threads 1`] = `
Immutable.List [
  Immutable.Map {
    participants: "Criptext Info <no-reply@criptext.com>",
    isMuted: 0,
    lastEmailId: "6Za2dcMlE0OSSc9",
    color: "#dbd230",
    delivered: 0,
    threadId: "6Za2dcMlE0OSSc9",
    timer: 1,
    preview: "v3m cNvAeKiY1EakWwnfTvLdO22YzkuS7bILjnWncQCwqHBHWfYNsQekd4A3",
    hasOpenAttachments: false,
    subject: "v3m cNvAeKiY1EakWwnfTvLdO22YzkuS7bILjnWncQCwqHBHWfYNsQekd4A3",
    totalAttachments: 1,
    letters: "CI",
    header: "Criptext Info",
    isTrash: 0,
    date: "ene. 20",
    timesOpened: 2,
    unread: true,
    status: 1,
    timestamp: 1516455734588,
    emails: Immutable.List [
      1,
    ],
    isDraft: 0,
    secure: 1,
    content: "v3m cNvAeKiY1EakWwnfTvLdO22YzkuS7bILjnWncQCwqHBHWfYNsQekd4A3",
    id: 18,
    selected: false,
    allowNotifications: 0,
    s3Key: "6Za2dcMlE0OSSc9",
    key: "6Za2dcMlE0OSSc9",
    labels: Immutable.Set [
      7,
      3,
      10,
      6,
    ],
  },
  Immutable.Map {
    participants: "Criptext Info <no-reply@criptext.com>",
    isMuted: 0,
    lastEmailId: "cuW6ElyoqsR7MMw",
    color: "#dbd230",
    delivered: 0,
    threadId: "cuW6ElyoqsR7MMw",
    timer: 1,
    preview: "Jq pUgbu1A0gywTYTeKuWfyfCNGgvWKnAK6SHx0ur 7qcMH3PU5S7ESoLTA3",
    hasOpenAttachments: false,
    subject: "Jq pUgbu1A0gywTYTeKuWfyfCNGgvWKnAK6SHx0ur 7qcMH3PU5S7ESoLTA3",
    totalAttachments: 1,
    letters: "CI",
    header: "Criptext Info",
    isTrash: 0,
    date: "ene. 20",
    timesOpened: 2,
    unread: true,
    status: 1,
    timestamp: 1516455734588,
    emails: Immutable.List [
      1,
    ],
    isDraft: 0,
    secure: 1,
    content: "Jq pUgbu1A0gywTYTeKuWfyfCNGgvWKnAK6SHx0ur 7qcMH3PU5S7ESoLTA3",
    id: 5,
    selected: false,
    allowNotifications: 0,
    s3Key: "cuW6ElyoqsR7MMw",
    key: "cuW6ElyoqsR7MMw",
    labels: Immutable.Set [
      7,
      3,
      10,
      6,
    ],
  },
]
`;

exports[`thread actions should add label 9 1`] = `
Immutable.List [
  Immutable.Map {
    participants: "Criptext Info <no-reply@criptext.com>",
    isMuted: 0,
    lastEmailId: "6Za2dcMlE0OSSc9",
    color: "#dbd230",
    delivered: 0,
    threadId: "6Za2dcMlE0OSSc9",
    timer: 1,
    preview: "v3m cNvAeKiY1EakWwnfTvLdO22YzkuS7bILjnWncQCwqHBHWfYNsQekd4A3",
    hasOpenAttachments: false,
    subject: "v3m cNvAeKiY1EakWwnfTvLdO22YzkuS7bILjnWncQCwqHBHWfYNsQekd4A3",
    totalAttachments: 1,
    letters: "CI",
    header: "Criptext Info",
    isTrash: 0,
    date: "ene. 20",
    timesOpened: 2,
    unread: true,
    status: 1,
    timestamp: 1516455734588,
    emails: Immutable.List [
      1,
    ],
    isDraft: 0,
    secure: 1,
    content: "v3m cNvAeKiY1EakWwnfTvLdO22YzkuS7bILjnWncQCwqHBHWfYNsQekd4A3",
    id: 18,
    selected: false,
    allowNotifications: 0,
    s3Key: "6Za2dcMlE0OSSc9",
    key: "6Za2dcMlE0OSSc9",
    labels: Immutable.Set [
      7,
      3,
      10,
      9,
    ],
  },
]
`;

exports[`thread actions should add threads to state 1`] = `
Immutable.List [
  Immutable.Map {
    participants: "Criptext Info <no-reply@criptext.com>",
    isMuted: 0,
    lastEmailId: "6Za2dcMlE0OSSc9",
    color: "#dbd230",
    delivered: 0,
    threadId: "6Za2dcMlE0OSSc9",
    timer: 1,
    preview: "v3m cNvAeKiY1EakWwnfTvLdO22YzkuS7bILjnWncQCwqHBHWfYNsQekd4A3",
    hasOpenAttachments: false,
    subject: "v3m cNvAeKiY1EakWwnfTvLdO22YzkuS7bILjnWncQCwqHBHWfYNsQekd4A3",
    totalAttachments: 1,
    letters: "CI",
    header: "Criptext Info",
    isTrash: 0,
    date: "ene. 20",
    timesOpened: 2,
    unread: true,
    status: 1,
    timestamp: 1516455734588,
    emails: Immutable.List [
      1,
    ],
    isDraft: 0,
    secure: 1,
    content: "v3m cNvAeKiY1EakWwnfTvLdO22YzkuS7bILjnWncQCwqHBHWfYNsQekd4A3",
    id: 18,
    selected: false,
    allowNotifications: 0,
    s3Key: "6Za2dcMlE0OSSc9",
    key: "6Za2dcMlE0OSSc9",
    labels: Immutable.Set [
      7,
      3,
      10,
    ],
  },
  Immutable.Map {
    participants: "Criptext Info <no-reply@criptext.com>",
    isMuted: 0,
    lastEmailId: "cuW6ElyoqsR7MMw",
    color: "#dbd230",
    delivered: 0,
    threadId: "cuW6ElyoqsR7MMw",
    timer: 1,
    preview: "Jq pUgbu1A0gywTYTeKuWfyfCNGgvWKnAK6SHx0ur 7qcMH3PU5S7ESoLTA3",
    hasOpenAttachments: false,
    subject: "Jq pUgbu1A0gywTYTeKuWfyfCNGgvWKnAK6SHx0ur 7qcMH3PU5S7ESoLTA3",
    totalAttachments: 1,
    letters: "CI",
    header: "Criptext Info",
    isTrash: 0,
    date: "ene. 20",
    timesOpened: 2,
    unread: true,
    status: 1,
    timestamp: 1516455734588,
    emails: Immutable.List [
      1,
    ],
    isDraft: 0,
    secure: 1,
    content: "Jq pUgbu1A0gywTYTeKuWfyfCNGgvWKnAK6SHx0ur 7qcMH3PU5S7ESoLTA3",
    id: 5,
    selected: false,
    allowNotifications: 0,
    s3Key: "cuW6ElyoqsR7MMw",
    key: "cuW6ElyoqsR7MMw",
    labels: Immutable.Set [
      7,
      3,
      10,
    ],
  },
]
`;

exports[`thread actions should remove label 7 1`] = `
Immutable.List [
  Immutable.Map {
    participants: "Criptext Info <no-reply@criptext.com>",
    isMuted: 0,
    lastEmailId: "6Za2dcMlE0OSSc9",
    color: "#dbd230",
    delivered: 0,
    threadId: "6Za2dcMlE0OSSc9",
    timer: 1,
    preview: "v3m cNvAeKiY1EakWwnfTvLdO22YzkuS7bILjnWncQCwqHBHWfYNsQekd4A3",
    hasOpenAttachments: false,
    subject: "v3m cNvAeKiY1EakWwnfTvLdO22YzkuS7bILjnWncQCwqHBHWfYNsQekd4A3",
    totalAttachments: 1,
    letters: "CI",
    header: "Criptext Info",
    isTrash: 0,
    date: "ene. 20",
    timesOpened: 2,
    unread: true,
    status: 1,
    timestamp: 1516455734588,
    emails: Immutable.List [
      1,
    ],
    isDraft: 0,
    secure: 1,
    content: "v3m cNvAeKiY1EakWwnfTvLdO22YzkuS7bILjnWncQCwqHBHWfYNsQekd4A3",
    id: 18,
    selected: false,
    allowNotifications: 0,
    s3Key: "6Za2dcMlE0OSSc9",
    key: "6Za2dcMlE0OSSc9",
    labels: Immutable.Set [
      10,
      3,
    ],
  },
]
`;

exports[`thread actions should set all threads as not selected 1`] = `
Immutable.List [
  Immutable.Map {
    participants: "Criptext Info <no-reply@criptext.com>",
    isMuted: 0,
    lastEmailId: "6Za2dcMlE0OSSc9",
    color: "#dbd230",
    delivered: 0,
    threadId: "6Za2dcMlE0OSSc9",
    timer: 1,
    preview: "v3m cNvAeKiY1EakWwnfTvLdO22YzkuS7bILjnWncQCwqHBHWfYNsQekd4A3",
    hasOpenAttachments: false,
    subject: "v3m cNvAeKiY1EakWwnfTvLdO22YzkuS7bILjnWncQCwqHBHWfYNsQekd4A3",
    totalAttachments: 1,
    letters: "CI",
    header: "Criptext Info",
    isTrash: 0,
    date: "ene. 20",
    timesOpened: 2,
    unread: true,
    status: 1,
    timestamp: 1516455734588,
    emails: Immutable.List [
      1,
    ],
    isDraft: 0,
    secure: 1,
    content: "v3m cNvAeKiY1EakWwnfTvLdO22YzkuS7bILjnWncQCwqHBHWfYNsQekd4A3",
    id: 18,
    selected: false,
    allowNotifications: 0,
    s3Key: "6Za2dcMlE0OSSc9",
    key: "6Za2dcMlE0OSSc9",
    labels: Immutable.Set [
      7,
      3,
      10,
    ],
  },
  Immutable.Map {
    participants: "Criptext Info <no-reply@criptext.com>",
    isMuted: 0,
    lastEmailId: "cuW6ElyoqsR7MMw",
    color: "#dbd230",
    delivered: 0,
    threadId: "cuW6ElyoqsR7MMw",
    timer: 1,
    preview: "Jq pUgbu1A0gywTYTeKuWfyfCNGgvWKnAK6SHx0ur 7qcMH3PU5S7ESoLTA3",
    hasOpenAttachments: false,
    subject: "Jq pUgbu1A0gywTYTeKuWfyfCNGgvWKnAK6SHx0ur 7qcMH3PU5S7ESoLTA3",
    totalAttachments: 1,
    letters: "CI",
    header: "Criptext Info",
    isTrash: 0,
    date: "ene. 20",
    timesOpened: 2,
    unread: true,
    status: 1,
    timestamp: 1516455734588,
    emails: Immutable.List [
      1,
    ],
    isDraft: 0,
    secure: 1,
    content: "Jq pUgbu1A0gywTYTeKuWfyfCNGgvWKnAK6SHx0ur 7qcMH3PU5S7ESoLTA3",
    id: 5,
    selected: false,
    allowNotifications: 0,
    s3Key: "cuW6ElyoqsR7MMw",
    key: "cuW6ElyoqsR7MMw",
    labels: Immutable.Set [
      7,
      3,
      10,
    ],
  },
]
`;

exports[`thread actions should set all threads as selected 1`] = `
Immutable.List [
  Immutable.Map {
    participants: "Criptext Info <no-reply@criptext.com>",
    isMuted: 0,
    lastEmailId: "6Za2dcMlE0OSSc9",
    color: "#dbd230",
    delivered: 0,
    threadId: "6Za2dcMlE0OSSc9",
    timer: 1,
    preview: "v3m cNvAeKiY1EakWwnfTvLdO22YzkuS7bILjnWncQCwqHBHWfYNsQekd4A3",
    hasOpenAttachments: false,
    subject: "v3m cNvAeKiY1EakWwnfTvLdO22YzkuS7bILjnWncQCwqHBHWfYNsQekd4A3",
    totalAttachments: 1,
    letters: "CI",
    header: "Criptext Info",
    isTrash: 0,
    date: "ene. 20",
    timesOpened: 2,
    unread: true,
    status: 1,
    timestamp: 1516455734588,
    emails: Immutable.List [
      1,
    ],
    isDraft: 0,
    secure: 1,
    content: "v3m cNvAeKiY1EakWwnfTvLdO22YzkuS7bILjnWncQCwqHBHWfYNsQekd4A3",
    id: 18,
    selected: true,
    allowNotifications: 0,
    s3Key: "6Za2dcMlE0OSSc9",
    key: "6Za2dcMlE0OSSc9",
    labels: Immutable.Set [
      7,
      3,
      10,
    ],
  },
  Immutable.Map {
    participants: "Criptext Info <no-reply@criptext.com>",
    isMuted: 0,
    lastEmailId: "cuW6ElyoqsR7MMw",
    color: "#dbd230",
    delivered: 0,
    threadId: "cuW6ElyoqsR7MMw",
    timer: 1,
    preview: "Jq pUgbu1A0gywTYTeKuWfyfCNGgvWKnAK6SHx0ur 7qcMH3PU5S7ESoLTA3",
    hasOpenAttachments: false,
    subject: "Jq pUgbu1A0gywTYTeKuWfyfCNGgvWKnAK6SHx0ur 7qcMH3PU5S7ESoLTA3",
    totalAttachments: 1,
    letters: "CI",
    header: "Criptext Info",
    isTrash: 0,
    date: "ene. 20",
    timesOpened: 2,
    unread: true,
    status: 1,
    timestamp: 1516455734588,
    emails: Immutable.List [
      1,
    ],
    isDraft: 0,
    secure: 1,
    content: "Jq pUgbu1A0gywTYTeKuWfyfCNGgvWKnAK6SHx0ur 7qcMH3PU5S7ESoLTA3",
    id: 5,
    selected: true,
    allowNotifications: 0,
    s3Key: "cuW6ElyoqsR7MMw",
    key: "cuW6ElyoqsR7MMw",
    labels: Immutable.Set [
      7,
      3,
      10,
    ],
  },
]
`;

exports[`thread actions should set thread as read 1`] = `
Immutable.List [
  Immutable.Map {
    participants: "Criptext Info <no-reply@criptext.com>",
    isMuted: 0,
    lastEmailId: "6Za2dcMlE0OSSc9",
    color: "#dbd230",
    delivered: 0,
    threadId: "6Za2dcMlE0OSSc9",
    timer: 1,
    preview: "v3m cNvAeKiY1EakWwnfTvLdO22YzkuS7bILjnWncQCwqHBHWfYNsQekd4A3",
    hasOpenAttachments: false,
    subject: "v3m cNvAeKiY1EakWwnfTvLdO22YzkuS7bILjnWncQCwqHBHWfYNsQekd4A3",
    totalAttachments: 1,
    letters: "CI",
    header: "Criptext Info",
    isTrash: 0,
    date: "ene. 20",
    timesOpened: 2,
    unread: false,
    status: 1,
    timestamp: 1516455734588,
    emails: Immutable.List [
      1,
    ],
    isDraft: 0,
    secure: 1,
    content: "v3m cNvAeKiY1EakWwnfTvLdO22YzkuS7bILjnWncQCwqHBHWfYNsQekd4A3",
    id: 18,
    selected: false,
    allowNotifications: 0,
    s3Key: "6Za2dcMlE0OSSc9",
    key: "6Za2dcMlE0OSSc9",
    labels: Immutable.Set [
      7,
      3,
      10,
    ],
  },
]
`;

exports[`thread actions should set thread as selected 1`] = `
Immutable.List [
  Immutable.Map {
    participants: "Criptext Info <no-reply@criptext.com>",
    isMuted: 0,
    lastEmailId: "6Za2dcMlE0OSSc9",
    color: "#dbd230",
    delivered: 0,
    threadId: "6Za2dcMlE0OSSc9",
    timer: 1,
    preview: "v3m cNvAeKiY1EakWwnfTvLdO22YzkuS7bILjnWncQCwqHBHWfYNsQekd4A3",
    hasOpenAttachments: false,
    subject: "v3m cNvAeKiY1EakWwnfTvLdO22YzkuS7bILjnWncQCwqHBHWfYNsQekd4A3",
    totalAttachments: 1,
    letters: "CI",
    header: "Criptext Info",
    isTrash: 0,
    date: "ene. 20",
    timesOpened: 2,
    unread: true,
    status: 1,
    timestamp: 1516455734588,
    emails: Immutable.List [
      1,
    ],
    isDraft: 0,
    secure: 1,
    content: "v3m cNvAeKiY1EakWwnfTvLdO22YzkuS7bILjnWncQCwqHBHWfYNsQekd4A3",
    id: 18,
    selected: true,
    allowNotifications: 0,
    s3Key: "6Za2dcMlE0OSSc9",
    key: "6Za2dcMlE0OSSc9",
    labels: Immutable.Set [
      7,
      3,
      10,
    ],
  },
]
`;
